src/util/Date.h

```diff
@@ -354,6 +355,22 @@ class DateOrLargeYear {
     bits_ = std::bit_cast<uint64_t>(d) | (datetime << numPayloadBits);
   }
 
+  // Construct a `DateOrLargeYear` given a `Timezone` object from the `Date`
+  // class. This Constructor sets the value of `timezone` (0) as the
+  // most significant bit. Thus compared with other underlying values a
+  // `DateOrLargeYear` can hold (`(large) positive year`, `(large) negative
+  // year` and `dateTime`), the here constructed value will always be smaller.
+  // (`DateOrLargeYear` compares w.r.t. year, we assume a duration constructed
+  // from a `TimeZone` doesn't hold a year)
+  // When comparing two `DateOrLargeYear` objects with two underlying
+  // (Date)TimeZone as duration values, we implicitly compare only w.r.t. the
+  // `Date::TimeZone` values, because all values except the timezone get some
+  // default values.
+  explicit DateOrLargeYear(Date::TimeZone tz) {
+    bits_ = std::bit_cast<uint64_t>(Date{0, 0, 0, -1, 0, 0.0, tz}) |
+            (timezone << numPayloadBits);
+  }
```

Comment:
This currently only supports full hours, but the 
`dayTimeDuration` type also supports seconds.
So you should choose a storage format, that internaly converts 
the duration to (fractions of) seconds, and then uses this as the payload)